File: backend/core/src/main/java/site/timecapsulearchive/core/domain/capsule/treasure_capsule/data/dto/TreasureCapsuleOpenDto.java
package site.timecapsulearchive.core.domain.capsule.treasure_capsule.data.dto;

import java.util.function.UnaryOperator;
import site.timecapsulearchive.core.domain.capsule.treasure_capsule.data.response.TreasureCapsuleOpenResponse;

public record TreasureCapsuleOpenDto(
    TreasureOpenStatus treasureOpenStatus,
    String treasureImageUrl
) {

    public TreasureCapsuleOpenResponse toResponse(
        final UnaryOperator<String> s3PreSignedUrlForGet) {
        String imageUrl = s3PreSignedUrlForGet.apply(treasureImageUrl);

        return new TreasureCapsuleOpenResponse(treasureOpenStatus, imageUrl);
    }
}

File: frontend/ARchive/app/src/main/java/com/droidblossom/archive/data/dto/treasure/response/TreasureCapsuleOpenDto.kt
package com.droidblossom.archive.data.dto.treasure.response

import com.droidblossom.archive.domain.model.treasure.response.TreasureCapsuleOpen
import com.droidblossom.archive.domain.model.treasure.response.TreasureOpenStatus

data class TreasureCapsuleOpenDto(
    val treasureOpenStatus: String,
    val treasureImageUrl: String
) {
    fun toModel() = TreasureCapsuleOpen(
        treasureOpenStatus = TreasureOpenStatus.fromDescription(this.treasureOpenStatus),
        treasureImageUrl = this.treasureImageUrl
    )
}